eras/conway/impl/src/Cardano/Ledger/Conway/UTxO.hs

```diff
@@ -118,7 +144,19 @@ conwayConsumedValue pp lookupKeyDeposit lookupDRepDeposit utxo txBody =
                 | Just deposit <- lookupDRepDeposit c -> (drepRegsInTx, totalRefund <+> deposit)
               _ -> accum
          in inject $ snd $ foldl' go (Set.empty, Coin 0) $ txBody ^. certsTxBodyL
-   in maryRefunds <+> drepRefunds
+      conwayRegRefunds =
```

Comment:
Yes, this approach is incorrect. The way you pointed out in the PR comment is the correct one. However, there is a much cleaner way to solve this without all the duplication

We can meet and talk about it.